src/kontrol/kdist/cheatcodes.md

```diff
@@ -426,7 +426,27 @@ This rule returns a symbolic boolean value being either 0 (false) or 1 (true).
          <k> #call_foundry SELECTOR _ => . ... </k>
          <output> _ => #buf(32, ?WORD) </output>
       requires SELECTOR ==Int selector ( "freshBool()" )
-       ensures #rangeBool(?WORD)
+       ensures #rangeBool(?WORD2)
+       [preserves-definedness]
+```
+
+#### `freshBytes` - Returns a single symbolic boolean.
+
+```
+function freshBytes(uint256) external returns (bytes memory);
+```
+
+`foundry.call.freshBytes` will match when the `freshBytes` cheat code function is called.
+This rule returns a symbolic boolean value being either 0 (false) or 1 (true).
+
+```{.k .symbolic}
+    rule [foundry.call.freshBytes]:
+         <k> #call_foundry SELECTOR ARGS => . ... </k>
+         <output> _ => Int2Bytes(32, 32, BE) +Bytes Int2Bytes(32, #asWord(ARGS), BE) +Bytes ?BYTES +Bytes
+         Int2Bytes(#asWord(ARGS) modInt 32, 0, BE) </output>
```

Comment:
`#buf(32, 32) +Bytes #buf(32, #asWord(ARGS)) +Bytes ?Bytes +Bytes #buf ( ( ( notMaxUInt5 &Int ( #asWord(ARGS) +Int maxUInt5 ) ) -Int #asWord(ARGS) ) , 0 )`

Reply:
The first term is needed because the format requires an offset to the beginning of the variable length byte array. I've fixed the padding and used `#buf` as suggested.